### PR TITLE
Fixes #7897: uses VS projects included with LCMS so that the recipe can build 32-bit lcms on windows 10 with VS2019

### LINKED ISSUE #7897: [package] lcms/2.12: can't build win32 lib with VS2019 on win10
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **lcms/2.12**
  * Operating System+version: **windows 10**
  * Compiler+version: **VS2019**
  * Docker image: NA
  * Conan version: **conan 1.41.0**
  * Python version: **Python 3.6.13 :: Anaconda custom (64-bit)**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
Configuration for profile default:

[settings]
os=Windows
os_build=Windows
arch=x86_64
arch_build=x86_64
compiler=Visual Studio
compiler.version=16
build_type=Release
[options]
[conf]
[build_requires]
[env]
```

### Steps to reproduce (Include if Applicable)
Running the command below fails to build lcms/2.12 due to my machine's lack of windows SDK 8.1.  My `conanfile.txt` is in the `build/` subfolder of the current dir.

```
>conan install build -s arch=x86 -s build_type=Release -s compiler="Visual Studio" -s compiler.runtime=MT -s compiler.version=16 --install-folder build/win/conan/win32/Release --build
```

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
Configuration:
[settings]
arch=x86
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.runtime=MT
compiler.version=16
os=Windows
os_build=Windows
[options]
[build_requires]
[env]

lcms/2.12: Forced build from source
conanfile.txt: Installing package
Requirements
    lcms/2.12 from 'conancenter' - Cache
Packages
    lcms/2.12:d4bb31912cc39810f01c804d7ba2f80b8bd35a3a - Build

Cross-build from 'Windows:x86_64' to 'Windows:x86'
Installing (downloading, building) binaries...
lcms/2.12: Configuring sources in C:\Users\atevet\.conan\data\lcms\2.12\_\_\source
Downloading 2.12.tar.gz completed [7231.36k]

lcms/2.12: Copying sources to build folder
lcms/2.12: Building your package in C:\Users\atevet\.conan\data\lcms\2.12\_\_\build\d4bb31912cc39810f01c804d7ba2f80b8bd35a3a
lcms/2.12: Generator txt created conanbuildinfo.txt
lcms/2.12: Aggregating env generators
lcms/2.12: Calling build()
**********************************************************************
** Visual Studio 2019 Developer Command Prompt v16.10.3
** Copyright (c) 2021 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x86'

Microsoft Visual Studio 2019 Version 16.10.3.
Copyright (C) Microsoft Corp. All rights reserved.
Upgrading project 'testbed'...
        Configuration 'Debug|Win32': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Debug|x64': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Release|Win32': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Release|x64': changing Platform Toolset to 'v142' (was 'v140').
Upgrading project 'tifficc'...
        Configuration 'Debug|Win32': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Debug|x64': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Release|Win32': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Release|x64': changing Platform Toolset to 'v142' (was 'v140').
Upgrading project 'lcms2_DLL'...
        Configuration 'Debug|Win32': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Debug|x64': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Release|Win32': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Release|x64': changing Platform Toolset to 'v142' (was 'v140').
Upgrading project 'linkicc'...
        Configuration 'Debug|Win32': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Debug|x64': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Release|Win32': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Release|x64': changing Platform Toolset to 'v142' (was 'v140').
Upgrading project 'transicc'...
        Configuration 'Debug|Win32': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Debug|x64': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Release|Win32': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Release|x64': changing Platform Toolset to 'v142' (was 'v140').
Upgrading project 'jpegicc'...
        Configuration 'Debug|Win32': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Debug|x64': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Release|Win32': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Release|x64': changing Platform Toolset to 'v142' (was 'v140').
Upgrading project 'tiffdiff'...
        Configuration 'Debug|Win32': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Debug|x64': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Release|Win32': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Release|x64': changing Platform Toolset to 'v142' (was 'v140').
Upgrading project 'lcms2_static'...
        Configuration 'Debug|Win32': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Debug|x64': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Release|Win32': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Release|x64': changing Platform Toolset to 'v142' (was 'v140').
Upgrading project 'psicc'...
        Configuration 'Debug|Win32': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Debug|x64': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Release|Win32': changing Platform Toolset to 'v142' (was 'v140').
        Configuration 'Release|x64': changing Platform Toolset to 'v142' (was 'v140').

Migration completed successfully, but some warnings were detected during migration.
For more information, see the migration report:
C:\Users\atevet\.conan\data\lcms\2.12\_\_\build\d4bb31912cc39810f01c804d7ba2f80b8bd35a3a\source_subfolder\Projects\VC2015\UpgradeLog.htm
Microsoft (R) Build Engine version 16.10.2+857e5a733 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Microsoft\VC\v160\Microsoft.Cpp.WindowsSDK.targets(46,5): error MSB8036: The Windows SDK version 8.1 was not found
. Install the required version of Windows SDK or change the SDK version in the project property pages or by right-clicking the solution and selecting "Retarget solution". [C:\Users\atevet
\.conan\data\lcms\2.12\_\_\build\d4bb31912cc39810f01c804d7ba2f80b8bd35a3a\source_subfolder\Projects\VC2015\lcms2_static\lcms2_static.vcxproj]
lcms/2.12: 
lcms/2.12: ERROR: Package 'd4bb31912cc39810f01c804d7ba2f80b8bd35a3a' build failed
lcms/2.12: WARN: Build folder C:\Users\atevet\.conan\data\lcms\2.12\_\_\build\d4bb31912cc39810f01c804d7ba2f80b8bd35a3a
ERROR: lcms/2.12: Error in build() method, line 109
        self._build_visual_studio()
while calling '_build_visual_studio', line 82
        msbuild.build("lcms2.sln", targets=[target], platforms={"x86": "Win32"}, upgrade_project=upgrade_project, properties=properties)
        ConanException: Error 1 while executing set "VSCMD_START_DIR=%CD%" && call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC/Auxiliary/Build/vcvarsall.bat" x86 && devenv "lcms2.sln" /upgrade && msbuild "lcms2.sln" /p:Configuration="Release" /p:UseEnv=true /p:Platform="Win32" /m:4 /target:lcms2_static /p:PlatformToolset="v142" /verbosity:minimal /p:ForceImportBeforeCppTargets="C:\Users\atevet\.conan\data\lcms\2.12\_\_\build\d4bb31912cc39810f01c804d7ba2f80b8bd35a3a\source_subfolder\Projects\VC2015\conan_build.props" /p:WholeProgramOptimization="false"
```
</details>

### PR DESCRIPTION
**Pull request by avitevet**
This is a fix in lcms/2.12 for #7897.

This PR has been reopened because the previous PR (#7935) encountered CI problems.

By using the project files included in the LCMS distribution, a 32-bit package can be built on windows 10 with VS2019 using the recipe.

---

- [x ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by madebr — DISMISSED** [review]

File: recipes/lcms/all/conanfile.py [hunk 1]
```diff
@@ -69,10 +69,16 @@ def _build_visual_studio(self):
...
+                vc_sln_subdir = "VC2015"
+            elif Version(self.settings.compiler.version) == "15":
+                vc_sln_subdir = "VC2017"
+            else:
+                vc_sln_subdir = "VC2019"
+
         with tools.chdir(os.path.join(self._source_subfolder, "Projects", vc_sln_subdir )):
             target = "lcms2_DLL" if self.options.shared else "lcms2_static"
-            upgrade_project = Version(self.settings.compiler.version) > "12"
+            upgrade_project = Version(self.settings.compiler.version) > "16"
```
Comment (madebr) [inline recipes/lcms/all/conanfile.py:81]: think you need to keep the upgrade.
I had a similar problem with another library (I don't remember which), which I could work around by removing some lines out of the vcx project.
Perhaps [this line](https://github.com/mm2/Little-CMS/blob/c65934f44aaf6cfc1a37f30f379bb76d5c891ec8/Projects/VC2019/lcms2_static/lcms2_static.vcxproj#L25), but I don't guarantee anything.

First, let's try to upgrade the project, before patching the build script.

Comment (avitevet) [reply recipes/lcms/all/conanfile.py]: Thanks for the recommendation - I'm adding the upgrade back in!

**Comment by avitevet** [comment]
@madebr I'm stuck here... the project can't be upgraded because the VS2017 and VS2019 IDE licenses are expired.  Is this expected?

**Comment by madebr** [comment]
Not really.
Expired licenses is a CI infrastructure problem.
Let's cc @jgsogo @danimtb .
These guys have access to the backend.

**Comment by avitevet** [comment]
> 
> 
> Not really. Expired licenses is a CI infrastructure problem. Let's cc @jgsogo @danimtb . These guys have access to the backend.

Thanks! @jgsogo & @danimtb the code I've written tries to upgrade VS2017 and VS2019 projects, but the upgrades fail because the licenses on the CI machines have expired.  Is there anything that can be done about this?

**Comment by SSE4** [comment]
> > Not really. Expired licenses is a CI infrastructure problem. Let's cc @jgsogo @danimtb . These guys have access to the backend.
> 
> Thanks! @jgsogo & @danimtb the code I've written tries to upgrade VS2017 and VS2019 projects, but the upgrades fail because the licenses on the CI machines have expired. Is there anything that can be done about this?

licenses should be actice again, I have restarted the build

**Review by SSE4 — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Pull request merged**